src/producto/producto.service.ts

```diff
@@ -0,0 +1,19 @@
+import { Model } from 'mongoose';
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Producto, ProductoDocument } from './schemas/producto.schema';
+import { CreateProductoDTO } from './dto/producto.dto';
+
+@Injectable()
+export class ProductoService {
+  constructor(@InjectModel(Producto.name) private productoModel: Model<ProductoDocument>) {}
+
+  async create(createProductoDTO: CreateProductoDTO): Promise<Producto> {
+    const createdProducto = new this.productoModel(createProductoDTO);
+    return createdProducto.save();
+  }
+
```

Comment:
Aquí es donde usamos los `HttpExceptions` para mandar el `bad request`
```suggestion

  async create(createProductoDTO: CreateProductoDTO): Promise<Producto> {
    try {
      const createdProducto = new this.productoModel(createProductoDTO);
      return createdProducto.save();
    } catch (error) {
      throw new BadRequestException('El mensaje que más te guste')
    }
  }

```
`BadRequestException` también se importa desde `nestjs/common`

Reply:
Aqui en service no manejo los errores